src/ol_infrastructure/applications/edx_notes/__main__.py

```diff
@@ -0,0 +1,201 @@
+openedx_version_tag = notes_config.get("openedx_version_tag")
+notes_server_tag = f"open-edx-notes-server-{env_name}"
+target_vpc = network_stack.require_output(target_vpc_name)
+
+aws_account = get_caller_identity()
+vpc_id = target_vpc["id"]
+notes_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["edx-notes-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+aws_config = AWSBase(
+    tags={
+        "OU": notes_config.require("business_unit"),
+        "Environment": env_name,
+        "Application": "open-edx-notes",
+        "Owner": "platform-engineering",
+        "openedx_version": openedx_version_tag,
+    }
+)
+
+notes_instance_role = iam.Role(
+    f"edx-notes-instance-role-{env_name}",
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    path=f"/ol-applications/open-edx-notes/{stack_info.env_prefix}/{stack_info.env_suffix}/",  # noqa: E501
+    tags=aws_config.tags,
+)
+
+notes_vault_policy = vault.Policy(
+    "edx-notes-vault-policy",
+    name=f"edx-notes-{stack_info.env_prefix}",
+    policy=Path(__file__)
+    .parent.joinpath("edx_notes_policy.hcl")
+    .read_text()
+    .replace("DEPLOYMENT", f"{stack_info.env_prefix}"),
+)
+aws_vault_backend = f"aws-{stack_info.env_prefix}"
+iam.RolePolicyAttachement(
+    f"edx-notes-describe-instance-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["describe_instances"],
+    role=notes_instance_role.name,
+)
+notes_instance_profile = iam.InstanceProfile(
+    f"edx-notes-instance-profile-{env_name}",
+    role=notes_instance_role.name,
+    path="/ol-infrastructure/open-edx-notes-server/profile/",
+)
+notes_vault_auth_role = vault.aws.AuthBackendRole(
+    "notes-web-ami-ec2-vault-auth",
+    backend=aws_vault_backend,
+    role="notes-server",
+    inferred_entity_type="ec2_instance",
+    inferred_aws_region=aws_config.region,
+    bound_iam_instance_profile_arns=[notes_instance_profile.arn],
+    bound_ami_ids=[notes_ami.id],
+    bound_account_ids=[aws_account.account_id],
+    bound_vpc_ids=[vpc_id],
+    token_policies=[notes_vault_policy.name],
+)
+
+consul_datacenter = consul_stack.require_output("datacenter")
+
+notes_security_group = ec2.SecurityGroup(
+    f"notes-server-security-group-{env_name}",
+    name=f"notes-server-{target_vpc_name}-{env_name}",
+    description="Access control for notes severs.",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=NOTES_SERVICE_PORT,
+            to_port=NOTES_SERVICE_PORT,
```

Comment:
This should just be using standard HTTP ports because we want to route through the Traefik proxy